includes/functions/functions_search.php

```diff
@@ -156,3 +156,46 @@ function zen_parse_search_string($search_str = '', &$objects = array()) {
 
     return false;
 }
+
+    function zen_build_keyword_where_clause($fields, $string)
+    {
+        global $db;
+        if (zen_parse_search_string(stripslashes($string), $search_keywords)) {
+            $where_str = " AND (";
+            for ($i = 0, $n = sizeof($search_keywords); $i < $n; $i++) {
+                switch ($search_keywords[$i]) {
+                    case '(':
+                    case ')':
+                    case 'and':
+                    case 'or':
+                        $where_str .= " " . $search_keywords[$i] . " ";
+                        break;
+                    default:
+                        $sql_add = " (";
+                        $first_field = true;
+                        foreach ($fields as $field_name) {
+                            if (!$first_field) {
+                                $sql_add .= ' OR ';
+                            }
+                            $first_field = false;
```

Comment:
Should this `$first_field=false` be inside the `if` above it?

Reply:
no.  that would not work.  we would never hit it.  `$first_field` would be set to true, and then we would never set it to false.  it has to be outside of the if statement and after it.

Reply:
Oh duh. Right 😊